src/components/modeler/Modeler.vue

```diff
@@ -580,17 +580,21 @@ export default {
       }
       return boundaryEvent;
     },
-    replaceDefinition(definition, boundaryEvent, process) {
-      const definitionIndex = process.get('flowElements').indexOf(definition);
-      process.flowElements[definitionIndex] = boundaryEvent;
-    },
     ensureCancelActivityIsAddedToBoundaryEvents(process) {
       this.getBoundaryEvents(process).forEach(definition => {
         const boundaryEvent = this.createBoundaryEvent(definition);
         definition.get('outgoing').forEach(outgoing => outgoing.set('sourceRef', boundaryEvent));
         this.replaceDefinition(definition, boundaryEvent, process);
       });
     },
+    replaceDefinition(definition, boundaryEvent, process) {
+      const definitionIndex = process.get('flowElements').indexOf(definition);
+      process.flowElements[definitionIndex] = boundaryEvent;
+      const boundaryEventDiagram = this.planeElements.find((diagram) => {
+        return diagram.bpmnElement === definition;
+      });
+      boundaryEventDiagram.bpmnElement = boundaryEvent;
+    },
```

Comment:
Every shape in the BPMN diagram has a "shape" object (`BPMNShape`, `BPMNEdge`, etc.) in bpmn-moddle that references the definition object. The issue was that when the boundary event definition was replaced, the diagram element still referenced the unused definition object from before.

This fix grabs the diagram object and updates its definition reference.